### PR TITLE
In Jetpack_Google_Analytics, use the static property for instance of Universal or Legacy analytics

### PR DESCRIPTION
**Pull request by douglas-johnson**
Fixes https://github.com/Automattic/jetpack/issues/19755

#### Changes proposed in this Pull Request:
* Updated the `__construct` function in `Jetpack_Google_Analytics` to use `self::$analytics` rather than `$analytics` so that the instance of either `Jetpack_Google_Analytics_Universal` or `Jetpack_Google_Analytics_Legacy` is stored in the static property `analytics`.

#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:
* Enable Google Analytics
* Using the global instance of `$jetpack_google_analytics` ensure that `$jetpack_google_analytics::$analytics` is an instance of one of the analytics classes rather than false.

### CODE REVIEW

**Review by jeherve — DISMISSED** [review]
Thanks for the contribution. You'll need to add a changelog entry for this PR. You can use [the Jetpack CLI tool](https://github.com/Automattic/jetpack/blob/master/docs/monorepo.md#first-time) to do so, by running the following commands:

`yarn install && yarn cli-setup`

You can then add a changelog entry like so:

`jetpack changelog add`

Another alternative is the following:

`cd projects/plugins/jetpack/ && composer install && vendor/bin/changelogger add`.

Thank you!

**Comment by samiff** [comment]
Hi @douglas-johnson 👋 

This is looking good to me, could you merge latest `master` into your branch please.

**Comment by douglas-johnson** [comment]
@samiff I tried just now, let me know if it worked.

**Review by samiff — APPROVED** [review]

File: projects/plugins/jetpack/modules/google-analytics/wp-google-analytics.php [hunk 1]
```diff
@@ -54,11 +54,10 @@ private function __construct() {
 		// At this time, we only leverage universal analytics when enhanced ecommerce is selected and WooCommerce is active.
 		// Otherwise, don't bother emitting the tracking ID or fetching analytics.js
 		if ( class_exists( 'WooCommerce' ) && Jetpack_Google_Analytics_Options::enhanced_ecommerce_tracking_is_enabled() ) {
-			$analytics     = new Jetpack_Google_Analytics_Universal();
-			$amp_analytics = new Jetpack_Google_AMP_Analytics();
-
+			self::$analytics = new Jetpack_Google_Analytics_Universal();
+			$amp_analytics   = new Jetpack_Google_AMP_Analytics();
```
Comment (samiff) [inline projects/plugins/jetpack/modules/google-analytics/wp-google-analytics.php:58]: Hey @douglas-johnson seems like the linter has a gripe about the unused `$amp_analytics` assignment. Could you change that to: `new Jetpack_Google_AMP_Analytics();`

**Review by samiff — CHANGES_REQUESTED** [review]

**Comment by samiff** [comment]
Internal diff: D61392-code

File: projects/plugins/jetpack/modules/google-analytics/wp-google-analytics.php (hunk 1 above)
Comment (douglas-johnson) [reply projects/plugins/jetpack/modules/google-analytics/wp-google-analytics.php]: @samiff Sure, done.

**Review by samiff — APPROVED** [review]

**Comment by samiff** [comment]
@douglas-johnson Thanks for the contribution! These changes should ship in Jetpack 9.8 on June 1st, 2021 👍

**Comment by kraftbj** [comment]
Internal reference: r225754-wpcom

**Pull request merged**